examples/direct_fidelity_estimation.py

```diff
@@ -370,10 +397,7 @@ def direct_fidelity_estimation(circuit: cirq.Circuit, qubits: List[cirq.Qid],
             noisy_simulator.simulate(circuit)).final_density_matrix
 
     trial_results: List[TrialResult] = []
-    for _ in range(n_trials):
-        # Randomly sample as per probability.
-        i = np.random.choice(len(pauli_traces), p=p)
-
+    for i in _sample_pauli_traces(pauli_traces, n_trials, n_clifford_trials):
```

Comment:
more pythonic, I think: have `_sample_pauli_traces` yield pauli traces and nix `i` (or use `enumerate` if you want it in `TrialResult`)

Reply:
Done, I think, although I did the enumerate in the `_sample_pauli_traces()` and not here because we want the index of the pauli trace pre-sampling, and putting enumerate here would get the index of the trace post-sampling.